### DISCUSSION TITLE
Argon 2 pre-release general discussion

### DISCUSSION THREAD

**Discussion by DervexDev** [argon-rbx/argon#22] [General]
#### If so, this is the perfect place to comment!

**Comment by AridAjd** [comment]
Is this supposed to happen? When I sync with rojo, it syncs the `Aegis` folder properly into the ReplicatedStorage but when I do it with argon it ports the `Aegis` folder inside the game.
![image](https://github.com/argon-rbx/argon/assets/118916772/01629903-0341-4cf9-8d4c-00fec3a0366b)

EDIT: Just found out that argon runs the `default.project.json` file even though I have specified `dev.project.json`

**Reply by DervexDev** [reply]
No, it's a bug but I just fixed it. Will release `pre4` version after I'm done with other stuff.

**Reply by DervexDev** [reply]
Fixed. See #23

**Comment by AridAjd** [comment]
??
![image](https://github.com/argon-rbx/argon/assets/118916772/180ae7ed-e215-42e3-8b2b-c115665676a4)

(Also you didn't publish the release as pre-release, just letting yk)

EDIT: You can't connect to other ports from the plugin other than `8000`

![image](https://github.com/argon-rbx/argon/assets/118916772/b290a006-d362-4798-8baf-f72424852fef)
![image](https://github.com/argon-rbx/argon/assets/118916772/0513fcfa-6674-45e5-89b4-c20c053f8eac)

EDIT 2: Seems like the port can only be changed from the settings
![image](https://github.com/argon-rbx/argon/assets/118916772/eea1a61c-95fa-400f-af1a-5a9cb438d69d)

**Reply by DervexDev** [reply]
> ![image](https://private-user-images.githubusercontent.com/118916772/315781819-180ae7ed-e215-42e3-8b2b-c115665676a4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTExMTcwMDQsIm5iZiI6MTcxMTExNjcwNCwicGF0aCI6Ii8xMTg5MTY3NzIvMzE1NzgxODE5LTE4MGFlN2VkLWUyMTUtNDJlMy04YjJiLWMxMTU2NjU2NzZhNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyMlQxNDExNDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZmE1ZGM0NjgyNTdhY2I5ZTdkYWEzZWM1ZjYyMDhmMjFhNDUzMjhmNjY0NGM5YjU3YjI0NjUzYTQ3NjhjNjhhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.jhRzKRcPSCqH8vU21tPO33qLloEhdkWB_rX4BX6Ye50)

That's because your `%localappdata%/Roblox/Plugins` folder does not exist. Argon should create one in this case though.

> (Also you didn't publish the release as pre-release, just letting yk)

Yep, that's intentional. In order to make Argon auto update feature work release has to be marked as latest.

> EDIT 2: Seems like the port can only be changed from the settings

Can confirm. The input box on the default page is broken.

Will fix all of this stuff soon!

**Reply by DervexDev** [reply]
Fixed both issues in `2.0.0-pre5`

**Comment by CompletedLoop** [comment]
I really dont understand the point of this since this is now just a Rojo clone with no advantages. The whole attraction to argon was to just connect it to a external text editor with no hassle since Rojo was too extra for what most people wanted and the fact argon easily worked with already created projects. I know the original was a buggy mess but it still should do what is always did.

**Reply by DervexDev** [reply]
> I really dont understand the point of this since this is now just a Rojo clone with no advantages.

Well you are actually very wrong. Argon now offers literally everything Rojo does with addition of:
* `studio` subcommand
* `exec` subcommand
* Customizable in every way possible (via global config and flags)
* Automatic installer and updater
* More performant than Rojo on macOS
* Better documentation
* Built-in support for `roblox-ts` and `wally`
* Custom project templates
* Support for automated formatting and linting
* Better two-way sync
* Uses more modern libraries
* Cleaner meta files
* and many many more!

> The whole attraction to argon was to just connect it to a external text editor with no hassle since Rojo was too extra for what most people wanted and the fact argon easily worked with already created projects.

This does not change! VSC user won't notice that anything changed and it will be still so easy to setup and use Argon as it was.


# Disclaimer!
Not everything I mentioned is available right now! But it will be with the official `2.0.0` release!

**Reply by AridAjd** [reply]
> * Built-in support for `roblox-ts` and `wally`

How does wally work built-in? Or is it not implemented yet?

**Reply by DervexDev** [reply]
Right now you can `init` project with wally.toml manifest using `--wally` flag. Additionally Argon will install Wally dependencies automatically if Packages folder is empty or does not exist at all but this hasn't been implemented yet.

**Comment by AridAjd** [comment]
Is `argon studio` broken? It says that it's opening studio but nothing else happens.
![image](https://github.com/argon-rbx/argon/assets/118916772/2f4da220-eb7c-44c0-ab4f-9c3a1ce33450)

**Reply by DervexDev** [reply]
Can't reproduce this issue on macOS nor on Windows (arm). Do you have custom install of Roblox Studio by any chance?

**Reply by AridAjd** [reply]
Nope, don't have any custom location or anything related to modif of Roblox studio. I'll try re-installing studio though.

**Reply by AridAjd** [reply]
Well that's weird. The problem still occurs. This is the correct path, right?
![image](https://github.com/argon-rbx/argon/assets/118916772/50184c13-8991-4050-8d41-2942aa5723fa)

**Reply by DervexDev** [reply]
Yeah, this is the correct path. I've seen that you started learning Rust so tomorrow I'm gonna send you simple **uncompiled** app to run on your computer, which will hopefully help us find where the problem is.

**Reply by DervexDev** [reply]
Okay so download this Cargo project: [studiotest.zip](https://github.com/argon-rbx/argon/files/14898913/studiotest.zip) and run it with `cargo run`.

If the Roblox Studio won't launch replace:
```rust
match Command::new(path).spawn() {
```

with:
```rust
match Command::new(path).output() {
```

and send me the output.

**Reply by AridAjd** [reply]
Yep, it launched this time.

**Reply by DervexDev** [reply]
But does the Argon launch Roblox Studio either?

**Reply by AridAjd** [reply]
Nope, tried it. It doesn't.

**Reply by DervexDev** [reply]
Well the code I sent you is literally ripped from Argon's source code. Could you send me the output anyways then?

**Reply by AridAjd** [reply]
Only prints out the path.
![image](https://github.com/argon-rbx/argon/assets/118916772/17d6d3e9-aa80-41b3-8347-2203fc086f19)

Well, I don't mind if the problem is just happening to me. It's better if it's not a problem for everyone else.

**Reply by DervexDev** [reply]
That's actually weird. Will try to mess around with this once I'm done with other stuff.

**Comment by DervexDev** [comment]
Closing this discussion in favor of [this one](https://github.com/orgs/argon-rbx/discussions/24) since Argon 2 has been officially released!
